More maintainable mitthesis.cls update

btsp6 opened the pull request.

btsp6 committed:
More maintainable mitthesis.cls update

bkettle commented on mitthesis.cls:
```diff
@@ -180,10 +180,10 @@
-\DeclareOption{vi}{\typeout{Course VI/VIII thesis style.}\advance\vithesis by1}
-\vithesis=0
+%% Pre-2023 theses are copyrighted to MIT instead of the author.
+\newcount\oldthesis
```
should we name this something more descriptive, like `mitowned`? It is maybe conceivable that someone might have MIT own their thesis in the future, or that MIT will change their policies again.

bkettle commented on mitthesis.cls:
```diff
@@ -441,8 +436,7 @@ reserved.\par\permission}}
   \def\baselinestretch{1}\large
-  \noindent\choosecase{\small Certified by:}\hspace{1.11cm}\parbox[t]{11cm}{\small #1 \\ #2 \\ Thesis Supervisor}}
-  %\signature{Certified by}{#1 \\ #2 \\ Thesis Supervisor}}
+  \signature{Certified by}{#1 \\ #2 \\ Thesis Supervisor}}
```
Should we maintain the use of `\choosecase` here (and in the other signature places)? It was present in the original thesis to allow people to make things uppercase, if they want.

bkettle replied:
This actually wasn't present in the original, I had misremembered (and not sure why I put it there in the earlier version).

bkettle commented on mitthesis.cls:
```diff
@@ -529,16 +520,10 @@ requirements for the \@degreeword\ of \\
 
-\def\signature#1#2{\par\noindent#1\dotfill\null\\*
-  {\raggedleft #2\par}}
+\def\signature#1#2{\normalsize\raggedright\parbox{7em}{#1:}\parbox[t]{\dimexpr\textwidth-7em\relax}{#2\par}}
```
If we want to keep using choosecase, it could also go in the definition of signature here.

As an aside, this signature definition is way cleaner. Thanks for taking the time to set that up.

bkettle commented:
It could be nice to update the screenshot in the README. Even though the changes are pretty much non-visual, it would be nice to update the picture with every code change.

edit: done

bkettle commented:
Thank you for the changes! This indeed looks much cleaner. Consider my comments above as nits, if you want me to merge it as-is, let me know.

bkettle committed:
update screenshot

btsp6 committed:
More descriptive "mitowned" documentclass option

for representing MIT copyrighted theses (instead of just calling this option "old")
